### DISCUSSION TITLE
Add TUS Support for Azure Storage Adapter

### DISCUSSION THREAD

**Discussion by w0kyj** [directus/directus#23756] [Feature Requests]
### Summary

TUS was added for local and S3 file storage adapters to allow resumable file upload in chunks.

Related Discussions & PRs:
https://github.com/directus/directus/discussions/22311
https://github.com/directus/directus/pull/22673

### Basic Example

Adding existing TUS feature support for resumable file uploads on Azure Storage Adapter

### Motivation

- Video file uploads which range in size from 100M to 2000M
- Azure WAF has a limit of 500M for request payload

### Detailed Design

Implement TUS on the Azure Storage Adapter

One option using tusd:
https://tus.io/blog/2021/08/10/tusd-azure-storage

### Must Have:
- Resumable file uploads in chunks to Azure Storage Adapter

### Should Have:
- 

### Could Have:
- Also add support for Google Storage Adapter

### Won't Have:
- 

### Drawbacks

TBD

### Alternatives

Use Local or S3 adapters, but not great options if hosting is Azure

### Adoption Strategy

Existing env variables: https://docs.directus.io/self-hosted/config-options.html#chunked-uploads
- additional variables if needed

Requires Directus node restarts

### Unresolved Questions

Does Azure support TUS?

**Comment by ComfortablyCoding** [comment]
Closing as this has been implemented via #23904
